bin/reth/src/node/mod.rs

```diff
@@ -97,7 +100,13 @@ impl Command {
         let mut config: Config =
             confy::load_path(&self.config).wrap_err("Could not load config")?;
         config.peers.connect_trusted_nodes_only = self.network.trusted_only;
-        if !self.network.trusted_peers.is_empty() {
+
+        // overrides the bootnodes with --bootnodes flag
+        if let Some(bootnodes) = &self.bootnodes {
+            bootnodes.iter().for_each(|peer| {
+                config.peers.trusted_nodes.insert(*peer);
+            })
```

Comment:
This is not entirely correct as it does not override the bootnodes. I would suggest adding a method to `Config` that takes `Option<Vec<NodeRecord>>`. Something like:

```rs
pub fn set_bootnodes(&mut self, enrs: Option<Vec<NodeRecord>>);
```

The config can be found here: https://github.com/paradigmxyz/reth/blob/main/bin/reth/src/config.rs

The method should set a field in `Config` that we then use here: https://github.com/paradigmxyz/reth/blob/bcbc3dacc799fa309f63d95593909c14eaf1a780/bin/reth/src/config.rs#L37

If `self.bootnodes` is some in that part of the code, we should use the specified bootnodes, otherwise we should fall back to `mainnet_nodes()`.

Hope that makes sense!